backend/data/src/main/kotlin/io/tolgee/component/automations/processors/slackIntegration/SlackSubscriptionProcessor.kt

```diff
@@ -0,0 +1,60 @@
+package io.tolgee.component.automations.processors.slackIntegration
+
+import io.tolgee.activity.ActivityService
+import io.tolgee.activity.data.ActivityType
+import io.tolgee.activity.projectActivityView.ProjectActivityViewByRevisionProvider
+import io.tolgee.api.IProjectActivityModelAssembler
+import io.tolgee.component.automations.AutomationProcessor
+import io.tolgee.model.automations.AutomationAction
+import io.tolgee.model.slackIntegration.EventName
+import io.tolgee.model.slackIntegration.SlackConfig
+import org.springframework.context.ApplicationContext
+import org.springframework.stereotype.Component
+
+@Component
+class SlackSubscriptionProcessor(
+  private val activityService: ActivityService,
+  private val activityModelAssembler: IProjectActivityModelAssembler,
+  private val slackExecutor: SlackExecutor,
+  private val applicationContext: ApplicationContext,
+) : AutomationProcessor {
+  override fun process(
+    action: AutomationAction,
+    activityRevisionId: Long?,
+  ) {
+    if (activityRevisionId == null) return
+
+    val view =
+      ProjectActivityViewByRevisionProvider(
+        applicationContext = applicationContext,
+        activityRevisionId,
+        onlyCountInListAbove = 10,
+      ).get() ?: return
+
+    val activityModel = activityModelAssembler.toModel(view)
+
+    val data = SlackRequest(activityData = activityModel)
+    val config = action.slackConfig ?: return
+    slackExecutor.setHelper(data = data, slackConfig = config)
+
+    when (activityModel.type) {
```

Comment:
These types can modify relevant data:
```
  UNKNOWN,
  SET_TRANSLATIONS,
  CREATE_KEY,
  COMPLEX_EDIT,
  IMPORT,
  BATCH_PRE_TRANSLATE_BY_TM,
  BATCH_MACHINE_TRANSLATE,
  AUTO_TRANSLATE,
  BATCH_CLEAR_TRANSLATIONS,
  BATCH_COPY_TRANSLATIONS,
  AUTOMATION, // possibly, since it's generic
```